htmd/parameterization/detect.py

```diff
@@ -262,66 +265,74 @@ def detectParameterizableDihedrals(molecule):
 
     >>> import os
     >>> from htmd.home import home
-    >>> from htmd.parameterization.ffmolecule import FFMolecule, FFTypeMethod
+    >>> from htmd.molecule.molecule import Molecule
+    >>> from htmd.parameterization.fftype import FFTypeMethod, fftype
     >>> from htmd.parameterization.detect import detectParameterizableDihedrals
+    >>> from htmd.parameterization.util import canonicalizeAtomNames
 
     Find the parameterizable dihedrals of glycol
     >>> molFile = os.path.join(home('test-param'), 'glycol.mol2')
-    >>> mol = FFMolecule(molFile)
+    >>> mol = Molecule(molFile, guess=('bonds', 'angles', 'dihedrals'))
     >>> detectParameterizableDihedrals(mol)
-    [[(2, 1, 0, 4), (1, 2, 3, 9)], [(0, 1, 2, 3)]]
+    [[(0, 1, 2, 3)], [(1, 2, 3, 9), (2, 1, 0, 4)]]
 
     Find the parameterizable dihedrals of ethanolamine
     >>> molFile = os.path.join(home('test-param'), 'ethanolamine.mol2')
-    >>> mol = FFMolecule(molFile, method=FFTypeMethod.GAFF2) # Does not work with CGenFF
+    >>> mol = Molecule(molFile, guess=('bonds', 'angles', 'dihedrals'))
+    >>> mol = canonicalizeAtomNames(mol)
+    >>> prm, mol = fftype(mol, method=FFTypeMethod.GAFF2) # Does not work with CGenFF
     >>> detectParameterizableDihedrals(mol)
-    [[(2, 1, 0, 4)], [(0, 1, 2, 3)], [(1, 2, 3, 9), (1, 2, 3, 10)]]
+    [[(0, 1, 2, 3)], [(1, 2, 3, 9), (1, 2, 3, 10)], [(2, 1, 0, 4)]]
 
     Find the parameterizable dihedrals of benzamidine
     >>> molFile = os.path.join(home('test-param'), 'benzamidine.mol2')
-    >>> mol = FFMolecule(molFile)
+    >>> mol = Molecule(molFile, guess=('bonds', 'angles', 'dihedrals'))
     >>> detectParameterizableDihedrals(mol)
-    [[(1, 0, 6, 12), (1, 0, 6, 13), (5, 0, 6, 12), (5, 0, 6, 13)], [(0, 6, 12, 16), (0, 6, 12, 17), (0, 6, 13, 14), (0, 6, 13, 15)]]
+    [[(0, 6, 12, 16), (0, 6, 12, 17), (0, 6, 13, 14), (0, 6, 13, 15)], [(1, 0, 6, 12), (1, 0, 6, 13), (5, 0, 6, 12), (5, 0, 6, 13)]]
 
     # Check if the atom swapping does not affect results
 
     Find the parameterizable dihedrals of chlorethene
     >>> molFile = os.path.join(home('test-param'), 'chlorethene_1.mol2')
-    >>> mol = FFMolecule(molFile)
+    >>> mol = Molecule(molFile, guess=('bonds', 'angles', 'dihedrals'))
+    >>> mol = canonicalizeAtomNames(mol)
+    >>> prm, mol = fftype(mol, method=FFTypeMethod.CGenFF_2b6)
     >>> detectParameterizableDihedrals(mol)
     [[(2, 1, 0, 4), (2, 1, 0, 5)]]
 
     Find the parameterizable dihedrals of chlorethene (with swapped atoms)
     >>> molFile = os.path.join(home('test-param'), 'chlorethene_2.mol2')
-    >>> mol = FFMolecule(molFile)
+    >>> mol = Molecule(molFile, guess=('bonds', 'angles', 'dihedrals'))
+    >>> mol = canonicalizeAtomNames(mol)
```

Comment:
No renaming and type guessing needed here

Reply:
rm-ed